freeqdsk/geqdsk.py

```diff
@@ -182,6 +183,163 @@
+
+    When creating an instance, if keyword arguments are used, they should be the
+    canonical names.
+
+    Attributes
+    ----------
+    comment:
+        Header comment
+    shot:
+        Header shot number
+    nx, nw, nr:
+        Number of radial points
+    ny, nh, nz:
+        Number of vertical points
+    rdim:
+        Width of computational domain in R direction, float [meter]
+    zdim:
+        Height of computational domain in Z direction, float [meter]
+    rcentr:
+        Reference value of R, float [meter]
+    rleft:
+        R at left (inner) boundary, float [meter]
+    zmid:
+        Z at middle of domain, float [meter]
+    rmagx, rmaxis:
+        R at magnetic axis (0-point), float [meter]
+    zmagx, zmaxis:
+        Z at magnetic axis (0-point), float [meter]
+    simagx, psi_axis:
+        Poloidal flux :math:`\psi` at magnetic axis, float [weber / radian]
+    sibdry, psi_boundary:
+        Poloidal flux :math:`\psi` at plasma boundary, float [weber / radian]
+    bcentr:
+        Vacuum toroidal magnetic field at rcentr, float [tesla]
+    cpasma, current:
+        Plasma current, float [ampere]
+    fpol:
+        Poloidal current function :math:`F(\psi)=RB_t`, 1D array [meter * tesla]
+    pres, pressure:
+        Plasma pressure :math:`p(\psi)`, 1D array [pascal]
+    ffprime:
+        :math:`FF'(\psi)`, 1D array [meter**2 * tesla**2 * radian / weber]
+    pprime:
+        :math:`p'(\psi)`, 1D array [pascal * radian / weber]
+    psi, f, psirz:
+         Poloidal flux :math:`\psi`, 2D array [weber / radian]
+    qpsi:
+        Safety factor :math:`q(\psi)`, 1D array [dimensionless]
+    nbdry, nbbbs:
+        Number of points in the boundary grid, int
+    nlim, limitr:
+        Number of points in the limiter grid, int
+    rbdry, rbbbs:
+        R of boundary points, 1D array [meter]
+    zbdry, zbbbs:
+        Z of boundary points, 1D array [meter]
+    rlim:
+        R of limiter points, 1D array [meter]
+    zlim:
+        Z of limiter points, 1D array [meter]
+
+    Examples
+    --------
+
+    Variables can be accessed with either the attribute dot syntax or ``dict``
+    item access. For example, the poloidal flux on the magnetic axis can be
+    accessed via any one of the following:
+
+    >>> gfile = GEQDSKFile(...)
+    >>> print(gfile["simagx"])
+    >>> print(gfile.simagx)
+    >>> print(gfile.psi_axis)
+
+    """
+
+    comment: str
+    shot: int
+    nx: int
+    ny: int
+    rdim: float
+    zdim: float
+    rcentr: float
+    rleft: float
+    zmid: float
+    rmagx: float
+    zmagx: float
+    simagx: float
+    sibdry: float
+    bcentr: float
+    cpasma: float
+    fpol: FloatArray
+    pres: FloatArray
+    ffprime: FloatArray
+    pprime: FloatArray
+    psi: FloatArray
+    qpsi: FloatArray
+    nbdry: int
+    nlim: int
+    rbdry: FloatArray | None = None
+    zbdry: FloatArray | None = None
+    rlim: FloatArray | None = None
+    zlim: FloatArray | None = None
```

Comment:
We'll have to set `requires-python = ">=3.10"` in `pyproject.toml` to use this notation

Reply:
Ah yes, I might just switch back to `Optional` for now then, although SPEC0 says it's actually almost time to drop support for 3.10 even!

Reply:
I'm counting the days until we can finally drop 3.9. It shouldn't be hard to fix, as `ruff check --select=UP --fix` will do it all in one go.